internal/pkg/cli/pipeline_init.go

```diff
@@ -166,26 +170,35 @@ func (o *initPipelineOpts) Validate() error {
 
 // Ask prompts for required fields that are not passed in and validates them.
 func (o *initPipelineOpts) Ask() error {
-	if o.repoURL != "" {
-		if err := o.validateURL(o.repoURL); err != nil {
-			return err
-		}
-	} else {
-		if err := o.selectURL(); err != nil {
+	if o.name == "" {
+		if err := o.askPipelineName(); err != nil {
 			return err
 		}
 	}
+	if err := validatePipelineName(o.name, o.appName); err != nil {
+		return err
+	}
 
-	if len(o.environments) > 0 {
-		if err := o.validateEnvs(); err != nil {
+	// TODO: validate not a duplicate
+
+	if o.repoURL == "" {
+		if err := o.selectURL(); err != nil {
 			return err
 		}
+	}
+	if err := o.validateURL(o.repoURL); err != nil {
```

Comment:
While this _is_ prettier than the `else`, in the other commands we were trying to avoid validating selected URLs because we already know they're valid. What do you think?

Reply:
👍 I think we do something like this in some commands:

```go
if err := o.askOrValidatePipelineName(); err != nil { ...}

func (o *initPipelineOpts) askOrValidatePipelineName() error {
  if o.name != nil {
      return validatePipelineName(o.name, o.appName)
   }
   // prompt for stuff
}
```